lib/casclient/frameworks/rails/filter.rb

```diff
@@ -84,7 +84,7 @@ def self.logout(controller, service = nil)
           referer = service || controller.request.referer
           st = controller.session[:cas_last_valid_ticket]
           controller.send(:reset_session)
-          controller.send(:redirect_to, client.logout_url(referer))
+          controller.send(:redirect_to, client.logout_url(referer), allow_other_host: true)
```

Comment:
Actually... we are doing something very bad, I think. :scream: We are accepting `referer`, which is INSECURE, and we redirect to that UNCONDITIONALLY (line 84), skipping all the protections, that helpful Rails people have put there. :scream: 

Do we need that, or can we at least build some whitelist for allowed domains? I'm almost sure that some future pen-test could detect, and then we would have to fix this in a hurry, instead of when we still have some time to think.

What do you think about some sanitization of the parameter to this redirect, guys?

_(It would be nice to add that to the specs, as soon as they are working, maybe)_

Reply:
https://trustfoundry.net/2016/08/23/referer-redirection-inconspicuous-danger/ - for reading

Reply:
@Arsen7 so what to do here?
i couldn't use logout method of authentifikator without adding `allow_other_host`

Reply:
Can we somehow validate that the `referer` parameter in this line belongs to us or is some of our trusted domains?

It may be a new ticket if this bug with not being logged out is more painful than the possible opening of (not the first one, though) open proxy, even if relatively hidden.

@stereosupersonic : What do you think about it?

Reply:
its ok